keras_vggface/models.py

```diff
@@ -408,35 +119,33 @@ def senet_identity_block(input_tensor, kernel_size,
 
     return m
 
+def preprocessing():
+    input_shape = (None, None, 3)
+    input = Input(shape=input_shape, batch_size=1)
+
+    x = ChannelReversal()(input)
+    x = Resizing(224, 224, interpolation='bilinear', name="Resize")(x)
+    output = DepthwiseNormalization([91.4953, 103.8827, 131.0912])(x)
+
+    model = Model(input, output, name='preprocessing')
+    return model
+
 
 def SENET50(include_top=True, weights='vggface',
             input_tensor=None, input_shape=None,
             pooling=None,
             classes=8631):
-    input_shape = _obtain_input_shape(input_shape,
-                                      default_size=224,
-                                      min_size=197,
-                                      data_format=K.image_data_format(),
-                                      require_flatten=include_top,
-                                      weights=weights)
-
-    if input_tensor is None:
-        img_input = Input(shape=input_shape)
-    else:
-        if not K.is_keras_tensor(input_tensor):
-            img_input = Input(tensor=input_tensor, shape=input_shape)
-        else:
-            img_input = input_tensor
-    if K.image_data_format() == 'channels_last':
-        bn_axis = 3
-    else:
-        bn_axis = 1
+    print("Using modified model")
```

Comment:
I'm not sure why this is removed - it seems like it's just taking some of the flexibility out .

Reply:
I removed it during debugging, I actually don't know what it is. I basically did a lot of treeshaking to make it more readable during debugging. I will add the things I removed now.

Reply:
Yup added it back in, thanks 🥭